UKHomeOffice/nmsw-ui
# Nmsw 908 pagination fix

# Ticket

[NMSW-908](https://jira.bics-collaboration.homeoffice.gov.uk/browse/NMSW-908)

----

## AC

**Given** I am using NMSW external UI
**And** I want to seamlessly switch between pages
**When** I interact with the pagination component
**Then** I should be able to switch between pages as before

----

## To test

Prerequisites : 

- Ensure you have submitted declaration of few voyages, 10 or 12 voyage declarations would be good for testing this feature.
- In `useGetDeclarations.jsx` update thhe API call to add `&per_page=1` after the `page_start` parameter. Feel free to change the `per_page` param as much as you wish to simulate real pagination use-case. By default since we are notting passing anything in, it will always be `50`. So I recommend changing this in order to test this feature properly. 

- run app
- navigate to /your-voyages
- > you should be presented with voyages and it should display the first page
- try to navigate to next page
- > you should be take to the next page
- try to navigate again to the previous page
- > you should be taken to the first page (i.e. the previous page of current)
- click on a specific page number
- > this should take you to the page you requested for
- navigate to the first page again
- > the previous button should not be displayed
- navigate to the last page 
- > the next button should not be displayed
- navigate to any random page
- > the pagination should display the page before and page after for ease of navigation

----

## Notes

N/A
